src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,97 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int DEFAULT_GROW = 2;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private double threshold = DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        Node<K,V> node = table[getIndex(key)];
+        if (node == null) {
+            table[getIndex(key)] = new Node<>(key, value);
+            size++;
+            return;
+        } else {
+            while (node.next != null) {
+                if (Objects.equals(node.key,key)) {
+                    node.value = value;
+                    return;
+                }
+                node = node.next;
+            }
+            if (Objects.equals(node.key,key)) {
+                node.value = value;
+                return;
+            }
+        }
+        node.next = new Node<>(key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> node = table[getIndex(key)];
+
+        if (node == null) {
+            return null;
+        }
+        while (node != null) {
+            if (Objects.equals(node.key,key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void resize() {
+        if (size == threshold) {
```

Comment:
This IF should be outside this method.
Name `resize()` tell us that we make resize. Without any conditions.